src/Event/Types.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Event;
+
+use ArrayIterator;
+use Iterator;
+use IteratorAggregate;
+
+final class Types implements IteratorAggregate
+{
+    private $types;
+
+    public function __construct(Type ...$types)
+    {
+        $this->types = $types;
```

Comment:
This allows for 0 types inside the class, which is probably not very useful.

Using this you can restrict to 1 or more, instead of 0 or more:

```
public function __construct(Type $type, Type ...$types)
{
    $this->types = array_merge([$type], $types);
}
```

Reply:
+1

Reply:
Adjusted, @rpkamp!